### DISCUSSION TITLE
Best practices to reuse the same build script across multiple projects

### DISCUSSION THREAD

**Discussion by Kralizek** [cake-build/cake#3224] [Q&A] [answered]
I have difference repositories (including a repository template) that use a `build.cake` file with the same content.

As the number of these repositories is going to increase, I wonder what is the best practice to reuse as much as possible.

Best regards.

**Comment by nils-a** [comment, marked as answer]
Generally you could leverage the [load directive](https://cakebuild.net/docs/writing-builds/preprocessor-directives#load-directive) to reference one or multiple common Cake files. There was also a [nice blog post](http://www.leerichardson.com/2021/02/he-one-thing-i-wish-id-known-before.html) about this some time ago.

If you have all projects tightly coupled, you could simply reference other files by using relative paths. One step more complex, you could create a repository to contain your common Cake files and use a git submodule (or equivalent) to "share" the files between projects.

My personal favorite would be to publish a custom nupgk and distribute the Cake file(s) to be consumed by other builds. A sample of that would be [Cake.Recipe](https://github.com/cake-contrib/Cake.recipe).

**Reply by Kralizek** [reply]
Thanks for the comment.

Following the Recipe sample,

If I were to
* pack the build.cake into a nuget package
* reference said package with the "#load" directive

Do I need any "code" in the concrete build.cake file? Or do I need something the RunTarget command?

**Reply by nils-a** [reply]
you still need something in the `RunTarget`.

**Reply by Kralizek** [reply]
Yes, if I could limit my `build.cake` file to be something like:

```cake
#load nuget:?package=MyRecipePackage&version=1.0.0

var target = Argument("Target", "Full");

RunTarget(target);
```

I would be perfectly happy.

**Reply by nils-a** [reply]
That example should work 🚀 
(Given `MyRecipePackage` is set up ok)

**Comment by pascalberger** [comment]
See [Reusing Builds](https://cakebuild.net/docs/writing-builds/reusing-builds) for different approaches how code can be shared across builds.
